subprojects/core-api/src/main/java/org/gradle/vcs/VcsMappings.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.vcs;
+
+import org.gradle.api.Action;
+import org.gradle.api.Incubating;
+import org.gradle.internal.HasInternalProtocol;
+
+/**
+ * In settings.gradle:
+ * <pre>
+ * vcsMappings {
+ *   withModule("group:name") {
+ *     from vcs(GitVcs) {
+ *         url = "..."
+ *     }
+ *   }
+ *   all { details ->
```

Comment:
I don't really understand the usecase this supports.

Reply:
It would let you do something like:

```
all { details ->
   if (details.requested.group == "com.example") {
      from vcs(GitVcs) {
         url = "http://github.com/com.example/${details.requested.name}"
      }
   }
}
```

But I could see dropping this for now because it opens up a can of worms if you let a single VersionControlSpec map to multiple Maven coordinates.

Reply:
Yeah. That seems like an exceedingly rare use-case. More likely we'd have an implementation of `VcsMapping` that knows how to generate a correct `VersionControlSpec` for some universe of `ComponentSelector`s